### PR TITLE
Dim--dim on x y

### PR DESCRIPTION
**Pull request by schaumb**

### CODE REVIEW

File: src/chart/generator/marker.cpp [hunk 1]
```diff
@@ -75,7 +76,12 @@ Marker::Marker(const Options &options,
 	    index,
 	    horizontal ? &mainId->value : subAxisId);
 
-	spacing.x = (horizontal || (lineOrCircle && !polar))
+	spacing.x = (horizontal || (lineOrCircle && !polar)
+	                || (channels.at(ChannelId::y).isDimension()
+	                    && channels.at(ChannelId::y).hasDimension()
```
Comment (simzer) [inline src/chart/generator/marker.cpp:81]: Why do we need hasDimension(), shouldnt isDimension be enought?

File: src/chart/generator/marker.cpp [hunk 2]
```diff
@@ -88,7 +94,12 @@ Marker::Marker(const Options &options,
 	    index,
 	    !horizontal ? &mainId->value : subAxisId);
 
-	spacing.y = (!horizontal || lineOrCircle)
+	spacing.y = (!horizontal || lineOrCircle
+	                || (channels.at(ChannelId::x).isDimension()
+	                    && channels.at(ChannelId::x).hasDimension()
```
Comment (simzer) [inline src/chart/generator/marker.cpp:99]: This expression is getting too complex. Maybe extract newly added portion to a local variable?

File: src/chart/generator/plot.cpp [hunk 3]
```diff
@@ -129,6 +129,11 @@ Buckets Plot::generateMarkers()
 	for (auto &&[ix, mid] : options->markersInfo)
 		map.emplace(mid, ix);
 
+	bool rectangleSpacing{true};
+	if (auto &&sp = style.plot.marker.rectangleSpacing;
+	    sp && sp->get().has_value())
+		rectangleSpacing = *sp->get() > 0.0;
+
```
Comment (simzer) [inline src/chart/generator/plot.cpp:136]: Why is this needed? Cannot we handle this only at rendering? Marker::spacing supposed to be a style independent factor

File: src/chart/options/options.cpp [hunk 4]
```diff
@@ -115,6 +115,9 @@ void Options::drilldownTo(const Options &other)
 	for (auto &&dim : other.getChannels().getDimensions())
 		if (!getChannels().isSeriesUsed(dim))
 			stackChannel.addSeries(dim);
+	if (stackChannel.isDimension()
+	    && geometry == ShapeType::rectangle)
+		this->align = Base::Align::Type::stretch;
```
Comment (simzer) [inline src/chart/options/options.cpp:120]: It smells a bit "workaroundy". Stretch does the work, but it's purpose is complety different. 
I feel like that instead, we would need an `bool spacing` member in `Channel` which gives controls to manually swich on/off the spacing.

File: src/chart/generator/marker.cpp (hunk 1 above)
Comment (schaumb) [reply src/chart/generator/marker.cpp:81]: isDimension() && !hasDimension means == empty(). When it is an empty channel, there should be no spacing

File: src/chart/generator/plot.cpp (hunk 3 above)
Comment (simzer) [reply src/chart/generator/plot.cpp:136]: After checking the test cases, we concluded that this is not needed.

File: src/chart/options/options.cpp (hunk 4 above)
Comment (schaumb) [reply src/chart/options/options.cpp:120]: I noted. I'll not rewrite it in this PR but handle it at the axis refactor.

**Review by simzer — APPROVED** [review]
